builds/fluent_bit_local.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x -e
+DESTDIR=$1
+mkdir -p $DESTDIR
+
+cd submodules/fluent-bit
+mkdir -p build
+cd build
```

Comment:
Will this leave a bunch of added files from git's perspective? Or are the existing .gitignore rules in fluent-bit broad enough to ignore everything we'd be generating? If it's the former then dealing with the constant state of "git thinks there's a bunch of untracked files" will be toilsome and I'd recommend moving the build folder somewhere outside the source tree.

Reply:
Everything inside `build` is ignored in Fluent Bit. https://github.com/fluent/fluent-bit/blob/master/.gitignore#L10

After running the local build, the tree in my fluent-bit submodule is still clean:
```
braydonk@bk:~/Git/ops-agent/submodules/fluent-bit$ git status
HEAD detached at 7de2c4522
nothing to commit, working tree clean
```